aten/src/ATen/native/mps/operations/BinaryOps.mm

```diff
@@ -36,6 +36,8 @@ void binaryOpTensor(const Tensor& self, const Tensor& other, const Scalar& alpha
   const bool is_self_scalar = self.dim() == 0;
   const bool is_other_scalar = other.dim() == 0;
 
+  output.resize_(at::infer_size(self.sizes(), other.sizes()));
```

Comment:
We need to this `resize_` only for the unstructured ops which are being added. We can make this conditional:
```
   auto new_size = at::infer_size(self.sizes(), other.sizes());
   if (!output.is_same_size(new_size)) {
        output.resize_(new_size);
   }
```